src/Authenticator.php

```diff
@@ -36,12 +36,13 @@ public function __construct(
     public function beginAuthentication(
         Uri                            $redirectUri,
         ProviderConfigurationInterface $provider,
+        bool                           $withDefaultScopes = true,
     ): AuthenticationUriBuilder {
         if ($redirectUri->getScheme() !== 'https') {
             throw new InsecureUriException('Redirect URI must use https.');
         }
 
-        $authenticationUriBuilder = new AuthenticationUriBuilder($provider, $redirectUri);
+        $authenticationUriBuilder = new AuthenticationUriBuilder($provider, $redirectUri, $withDefaultScopes);
```

Comment:
What if instead you added a method to the `AuthenticationUriBuilder` called `withoutScopes()` that lets you remove scopes from the array? Then you could do:
```
$authenticationRedirectUri = $authenticator
    ->beginAuthentication($providerConfiguration, $redirectUrl)
    ->withoutScopes('openid')
    ->withScopes('default')
    ->uri();
```